src/mpi/request/wait.c

```diff
@@ -45,9 +45,6 @@ int MPIR_Wait_impl(MPIR_Request * request_ptr, MPI_Status * status)
                 (request_ptrs[i]->u.ureq.greq_fns->poll_fn) (request_ptrs[i]->u.ureq.
                                                              greq_fns->grequest_extra_state,
                                                              &array_of_statuses[i]);
-        }
-        if (request_ptr->kind == MPIR_REQUEST_KIND__GREQUEST &&
-            request_ptr->u.ureq.greq_fns->wait_fn != NULL) {
```

Comment:
This might be a stupid question, but is there a reason we wouldn't call `wait_fn` in this function if it exists?

Reply:
Actually I would like to know the answer too, in almost all cases `poke` i.e. the `poll_fn` version is used except for `Waitall` which uses the `wait_fn`. I didn't dare to change this behavior.